### DISCUSSION TITLE
Rendering of floating polygons that overlay the map, for UI elements

### DISCUSSION THREAD

**Discussion by daseinlaser** [maplibre/maplibre-native#1795] [General]
Hello,
Are there any ways to use Maplibre-native (Android) to overlay floating polygons over the displayed map? For example a rectangle, drawn over the map in a fixed location relative to the screen. If not, I plan to either extend Maplibre-native with such functionality, or to overlay the Maplibre MapView with my own UI view in my application. I am pretty sure that one can draw polygons on the map using Maplibre-native, but I believe that those are positioned according to location within the map.

**Comment by stefankarschti** [comment]
Hi! I'm working on enabling the users to emit custom `drawables` in _custom drawable map layers_. Various primitive types can be created, I started with polylines. Matrix is arbitrary, therefore you could draw polygons relative to the viewport if that's what's needed.
CC @louwers @sjg-wdw 
WIP PR: #1754
Check out issue #1735, comments welcome!
